File: examportal-backend/src/main/java/com/examportal/examportalbackend/service/QuizService.java
package com.examportal.examportalbackend.service;

import java.util.List;

import com.examportal.examportalbackend.dao.QuizRepo;
import com.examportal.examportalbackend.entity.exam.Category;
import com.examportal.examportalbackend.entity.exam.Quiz;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class QuizService {

    @Autowired
    private QuizRepo quizRepo;

    // add quiz
    public Quiz addQuiz(Quiz quiz) {
        return this.quizRepo.save(quiz);
    }

    // update quiz
    public Quiz updateQuiz(Quiz quiz) {
        return this.quizRepo.save(quiz);
    }

    // get all quizzes
    public List<Quiz> getAllQuizzes() {
        return this.quizRepo.findAll();
    }

    // get quiz by id
    public Quiz getQuizById(Long quizId) {
        return this.quizRepo.findById(quizId).get();
    }

    // delete quiz by id
    public void deleteQuiz(Long quizId) {
        this.quizRepo.deleteQuiz(quizId);
    }

    // get quizzes by category
    public List<Quiz> getQuizzesOfCategory(Category category){
        return this.quizRepo.findBycategory(category);
    }


    public List<Quiz> getActiveQuizzes(){
        return this.quizRepo.findByActive(true);
    }

    public List<Quiz> getActiveQuizzesOfCategory(Category category){
        return this.quizRepo.findByCategoryAndActive(category, true);
    }



}

File: examportal-frontend/src/app/services/quiz-service/quiz.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from '../helper';

@Injectable({
  providedIn: 'root'
})
export class QuizService {

  constructor(private http: HttpClient) { }

  // all quizzes
  public allQuizzes() {
    return this.http.get(`${baseUrl}/quiz/`);
  }

  // add quiz
  public addQuiz(quiz:any){
    return this.http.post(`${baseUrl}/quiz/`,quiz);
  }

  // delete quiz
  public deleteQuiz(id:any){
    return this.http.delete(`${baseUrl}/quiz/`+id);
  }

  // get single quiz
  public getQuiz(id:any){
    return this.http.get(`${baseUrl}/quiz/${id}`);
  }

  // update quiz
  public updateQuiz(quiz:any){
    return this.http.put(`${baseUrl}/quiz/`,quiz);
  }

  // get quizzes by category
  public getQuizzesOfCategory(cid:any){
    return this.http.get(`${baseUrl}/quiz/category/${cid}`);
  }

  // get active quizzes
  public getActiveQuizzes(){
    return this.http.get(`${baseUrl}/quiz/active`);
  }

  // get active quizzes of category
  public getActiveQuizOfCategory(cid:any){
    return this.http.get(`${baseUrl}/quiz/category/active/${cid}`);
  }

}
